libraries/testing/include/eosio/testing/tester.hpp

```diff
@@ -545,8 +575,8 @@ namespace eosio { namespace testing {
 
       using base_tester::produce_block;
 
-      signed_block_ptr produce_block( fc::microseconds skip_time = fc::milliseconds(config::block_interval_ms) )override {
-         return _produce_block(skip_time, false);
+      produce_block_result_t produce_block( fc::microseconds skip_time = fc::milliseconds(config::block_interval_ms), bool no_throw = false )override {
+         return _produce_block(skip_time, false, no_throw);
```

Comment:
So that all the existing tests, especially all the external user tests that expect `produce_block` to return the `signed_block_ptr`. I think you should add a new `produce_block_ex` or `produce_block_v2` or `block_produce` or `produce_block_traces` that returns the `produce_block_result_t`.

Reply:
Excellent point!